### PR TITLE
LPS-116954 portal-search-tuning-gsearch-configuration-web: Move useRef, use onBlur

### PR DESCRIPTION
**Pull request by oliv-yu**
Ok! Hopefully this is more straightforward! 🤞 UseRef and onBlur are now in AceEditor, when onBlur gets called the state gets updated in Fragment.

### CODE REVIEW

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js [hunk 1]
```diff
@@ -161,16 +146,14 @@ export default function Fragment({
...
-						}}
-						value={value}
-					/>
-				</div>
+				<AceEditor
+					onBlur={(value) => setValue(value)}
+					onRender={({editorElement, editorTextInput}) => {
+						editorElementRef.current = editorElement;
+						editorTextInputRef.current = editorTextInput;
+					}}
```
Comment (thektan) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js:154]: Is this needed anymore?

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js [hunk 2]
```diff
@@ -161,16 +146,14 @@ export default function Fragment({
...
+					value={value}
```
Comment (thektan) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js:155]: What if we got rid of the whole `value` state and this was `value={valueRef.current}` instead? (Meaning adding onChange back again)

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js (hunk 1 above)
Comment (oliv-yu) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js:154]: Oh I took that from FieldMappings, I guess it's not necessary so I'll take it out!

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js (hunk 2 above)
Comment (oliv-yu) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js:155]: Oh ok I see what you mean now, my bad, 😭 looks like it works with just `valueRef.current`. I'll add it in!

**Comment by oliv-yu** [comment]
😅 In the second to most recent commit was my attempt at getting Ace Editor working, the most recent commit is switching to using textarea and adding appropriate styling to make it look more convincing (it doesn't tab when you 'tab' in the textarea, but maybe we can work on that later if we choose to keep textarea!). 

For the json edits, since text area would accept and be edited as a string, the value is actually saved as 'jsonString' in the object of fragments. So for the form submission, I think we would need to do something like `JSON.parse(jsonString)` first and validate that it's acceptable JSON.

Thanks!

**Comment by thektan** [comment]
The backend will do the JSON parsing and validating so keeping it as a string will be fine. :)

**Pull request merged**